### PR TITLE
Add ExternalPropertyProvider

### LINKED ISSUE #6: Add ExternalPropertyProvider

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #6

### CODE REVIEW

File: src/main/java/org/kiwiproject/config/ExternalPropertyProvider.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+@Slf4j
+public class ExternalPropertyProvider implements ConfigProvider {
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/config/ExternalPropertyProvider.java:18]: Going to write docs later?

File: src/test/resources/ExternalPropertyProvider/config.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+unit.test.foo=bar
```
Comment (sleberknight) [inline src/test/resources/ExternalPropertyProvider/config.properties:1]: Back at you with the "no newline" crapola at end.... :-)

**Review by sleberknight — APPROVED** [review]

File: src/main/java/org/kiwiproject/config/ExternalPropertyProvider.java (hunk 1 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/config/ExternalPropertyProvider.java:18]: Doing it now

File: src/test/resources/ExternalPropertyProvider/config.properties (hunk 2 above)
Comment (chrisrohr) [reply src/test/resources/ExternalPropertyProvider/config.properties:1]: he he

**Review by sleberknight — APPROVED** [review]

**Pull request merged**